plugins/copr.py

```diff
@@ -223,6 +223,9 @@ def run(self):
                   'copr command are required'))
         try:
             chroot = self.opts.arg[1]
+            if len(self.opts.arg) > 2:
+                self.cli.optparser.print_help(self)
+                raise dnf.cli.CliError(_('too many arguments, a maximum of two arguments are allowed'))
```

Comment:
Looking at the referenced bug: https://bugzilla.redhat.com/show_bug.cgi?id=1801017

This doesn't solve the original traceback in the report:
```
# dnf copr enable kwizart/fedy foo
...
Do you really want to enable copr.fedorainfracloud.org/kwizart/fedy? [y/N]: y
Traceback (most recent call last):
  File "/home/lu/dev/dnf/bin/dnf-3", line 58, in <module>
    main.user_main(sys.argv[1:], exit_code=True)
  File "/home/lu/dev/dnf/dnf/cli/main.py", line 201, in user_main
    errcode = main(args)
  File "/home/lu/dev/dnf/dnf/cli/main.py", line 67, in main
    return _main(base, args, cli_class, option_parser_class)
  File "/home/lu/dev/dnf/dnf/cli/main.py", line 106, in _main
    return cli_run(cli, base)
  File "/home/lu/dev/dnf/dnf/cli/main.py", line 122, in cli_run
    cli.run()
  File "/home/lu/dev/dnf/dnf/cli/cli.py", line 1167, in run
    return self.command.run()
  File "/home/lu/dev/dnf-plugins-core/plugins/copr.py", line 273, in run
    self._download_repo(project_name, repo_filename, chroot)
  File "/home/lu/dev/dnf-plugins-core/plugins/copr.py", line 463, in _download_repo
    arch = chroot.split('-')[2]
IndexError: list index out of range
```

The problem is two-fold, one thing is checking for the correct number of arguments, the other is validation of the chroot argument being a valid chroot.

Reply:
I know, unfortunately, the format of the chroot is not defined anywhere, so in the end, I just add a check on the number of arguments.

Reply:
Okay, but you've got this error here which is causing a traceback, and the problem is the code expects the input to be in a certain format, which it isn't. So instead of a traceback you should print a message for this case, it's up to you how specific you want to be in the message - e.g. you can just say "invalid chroot format", or preferably state some details about the necessary format, it appears to be `distro-version-arch`.

So, IMO a good solution here is to split the chroot by `'-'` and verify the resulting list has exactly 3 items, because that's what the code expects. If it isn't, print an error message.

Reply:
> So, IMO a good solution here is to split the chroot by '-' and verify the resulting list has exactly 3 items, because that's what the code expects. If it isn't, print an error message.

+1, long-term we should fix the copr plugin and copr base code to allow artificial strings there (at least mock supports it, see e.g. mock-rpmfusion-free.rpm chroots).  But at this moment copr.py plugin seems to indeed strictly expect three parts in the chroot name so we should prevent the tracebacks.

Reply:
I added a control to the chroot format. Now we expect the format distribution-version-architecture, with anything else an error message will be displayed.